ecs-agent/tmds/server.go

```diff
@@ -0,0 +1,155 @@
+// copyright amazon.com inc. or its affiliates. all rights reserved.
+//
+// licensed under the apache license, version 2.0 (the "license"). you may
+// not use this file except in compliance with the license. a copy of the
+// license is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. this file is distributed
+// on an "as is" basis, without warranties or conditions of any kind, either
+// express or implied. see the license for the specific language governing
+// permissions and limitations under the license.
+package tmds
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/didip/tollbooth"
+	"github.com/gorilla/mux"
+
+	"github.com/aws/amazon-ecs-agent/ecs-agent/logger/audit"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/logger/audit/request"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/tmds/logging"
+	muxutils "github.com/aws/amazon-ecs-agent/ecs-agent/tmds/utils/mux"
+)
+
+const (
+	// TMDS IP and port
+	IPv4 = "127.0.0.1"
+	Port = 51679
```

Comment:
nit: would it make sense to reuse this port defined in the config? https://github.com/aws/amazon-ecs-agent/blob/782948476da6d995ad33c6a04130f8172820af27/agent/config/config.go#L48

I imagine this would be used in multiple places, also outside of TMDS?

Reply:
The plan is to remove the port constant in agent module and use `tmds.Port` imported from ecs-agent module.

Reply:
Also ecs-agent module cannot depend on agent module as that would be a circular dep...

Reply:
synced offline, this will be looked into outside of this PR's scope.